var/spack/repos/builtin/packages/mathematica/package.py

```diff
@@ -18,14 +18,38 @@ class Mathematica(Package):
        http://spack.readthedocs.io/en/latest/mirrors.html"""
 
     homepage = "https://www.wolfram.com/mathematica/"
-    url = 'file://{0}/Mathematica_12.0.0_LINUX.sh'.format(os.getcwd())
 
     version('12.0.0', sha256='b9fb71e1afcc1d72c200196ffa434512d208fa2920e207878433f504e58ae9d7',
             expand=False)
 
+    # Licensing
+    license_required = True
+    license_comment  = '#'
+    license_files    = ['Configuration/Licensing/mathpass']
+    license_url      = 'https://reference.wolfram.com/language/tutorial/RegistrationAndPasswords.html#857035062'
+
+    def url_for_version(self, version):
+        return "file://{0}/Mathematica_{1}_LINUX.sh".format(os.getcwd(), version)
+
     def install(self, spec, prefix):
         sh = which('sh')
         sh(self.stage.archive_file, '--', '-auto', '-verbose',
            '-targetdir={0}'.format(prefix),
            '-execdir={0}'.format(prefix.bin),
            '-selinux=y')
+        # .spack directory in the install directory may not exist and needs to be created
```

Comment:
This shouldn't be the case. If it is, it's a bug that should be handled in Spack core. Can you open an issue about this with steps to reproduce it?